src/views/Pools/components/Vault/VaultRoiCalculatorModal.tsx

```diff
@@ -23,22 +23,30 @@ export const VaultRoiCalculatorModal = ({
     },
   } = useVaultPoolByKey(pool.vaultKey)
 
-  const { getLockedApy } = useVaultApy()
+  const { getLockedApy, flexibleApy } = useVaultApy()
   const { t } = useTranslation()
 
   const [cakeVaultView, setCakeVaultView] = useState(initialView || 0)
 
   const [duration, setDuration] = useState(weeksToSeconds(1))
 
-  const buttonMenu = [
-    <ButtonMenuItem>{t('Flexible')}</ButtonMenuItem>,
-    maxLockDuration && maxLockDuration.gt(0) && <ButtonMenuItem>{t('Locked')}</ButtonMenuItem>,
-  ].filter(Boolean)
+  const buttonMenuItems = useMemo(
+    () =>
+      [
+        <ButtonMenuItem key="Flexible">{t('Flexible')}</ButtonMenuItem>,
+        maxLockDuration && maxLockDuration.gt(0) && <ButtonMenuItem key="Locked">{t('Locked')}</ButtonMenuItem>,
```

Comment:
Condition can be shortened to `maxLockDuration?.gt(0)`

Reply:
Done